### ISSUE TITLE
Select multiple videos on the watch history page

### ISSUE THREAD

**Issue by ghost** [TeamNewPipe/NewPipe#11490]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Only one video can be selected at a time on the existing page

When users want to share, add to playlist, delete and other operations at the same time, they need to repeat a lot of steps.

If a "multi-select" function can be added to the long press menu, multiple videos can be selected at one time and the operations required by the user can be completed.

![Screenshot_2024-08-31-18-14-59-609_org schabi newpipe](https://github.com/user-attachments/assets/f0ecb8b4-4ea5-4ed5-8416-f6fc633025d2)
![Screenshot_2024-08-31-18-23-50-843_org schabi newpipe](https://github.com/user-attachments/assets/44c2aaf0-a5b5-4266-af3b-7e00923841ce)

Just like file management. Multiple selection of files or folders
![Screenshot_2024-08-31-18-16-58-595_com mi android globalFileexplorer](https://github.com/user-attachments/assets/0e5096e0-ce7c-40ac-a4e2-ea2228f8e84c)

### Why do you want this feature?

Simplify user operation steps and make Newpipe easier to use.

**Comment by GfEW** [comment]
Selecting, then applying available menu actions to, multiple media at once would greatly improve larger operations in (local) playlists, too. It could even be useful in search results, channels, public playlists etc.

Therefore, it might make sense to implement multi-selection in a more general manner so it's available everywhere (i. e. in all media listings), not limited to history.

**Comment by vomindoraan** [comment]
Agreed with @GfEW. Multiple selection in playlists is definitely something that I need on a daily basis, and I've been manually working around it for years. It'd be really nice to not have to do that anymore.

**Comment by Stypox** [comment]
Closing in favor of https://github.com/TeamNewPipe/NewPipe/issues/12090
